### PR TITLE
Replace arrays of primitive types on std arrays

### PR DESCRIPTION
**Pull request by vmishenev**
According to https://github.com/Kotlin/dokka/issues/2041#issuecomment-887444062

### CODE REVIEW

**Comment by vmishenev** [comment]
I don't replace parameter types into `Callable` of `DRI`

File: plugins/base/src/main/kotlin/transformers/documentables/KotlinArrayDocumentableReplacerTransformer.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                    true)
+                            else -> null
+                        }
+                    }
+                }
+            }
+            ?: super.processGenericTypeConstructor(genericTypeConstructor)
+
+    override fun processModule(module: DModule): AnyWithChanges<DModule>  =
+        if(module.isJVM())
```
Comment (MarcinAman) [inline plugins/base/src/main/kotlin/transformers/documentables/KotlinArrayDocumentableReplacerTransformer.kt:60]: I think that we can narrow down the scope of searching only to java classes, but watch for mixed packages (where java and kotlin classes are in 1 package).

The easiest way to check language would be to look at the source like i did here: https://github.com/Kotlin/dokka/pull/2048/files

File: plugins/base/src/test/kotlin/filter/KotlinArrayDocumentableReplacerTest.kt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        """.trimMargin(),
+            configuration
+        ) {
+            preMergeDocumentablesTransformationStage = {
+                val params = it.firstOrNull()?.packages?.firstOrNull()?.functions?.firstOrNull()?.parameters
+
+                val typeArrayNames = listOf("IntArray", "BooleanArray", "FloatArray", "DoubleArray", "LongArray", "ShortArray",
+                    "CharArray", "ByteArray")
+
+                Assertions.assertEquals(typeArrayNames.size, params?.size)
```
Comment (MarcinAman) [inline plugins/base/src/test/kotlin/filter/KotlinArrayDocumentableReplacerTest.kt:42]: You don't need to prefix those with `Assertions`

**Review by MarcinAman — APPROVED** [review]
I think it looks ok, some minor comments. 

One thing that i am concerned about is what is the runtime penalty of such transformer. Could you check it out on some bigger repository like `ktor`?

**Comment by vmishenev** [comment]
I ran dokka on `ktor` without- and with such transformer 4 times. I don't see a significant difference. But my  measurements are very scattered. 
image

**Pull request merged**